src/version_specifier.cpp

```diff
@@ -86,4 +86,42 @@ auto wlcs::AtLeastVersion::describe() const -> std::string
     return std::string{">= "} + std::to_string(version);
 }
 
+wlcs::LessThanVersion::LessThanVersion(uint32_t version) noexcept
+    : version{version}
+{
+}
+
+auto wlcs::LessThanVersion::select_version(
+    uint32_t max_available_version,
+    uint32_t max_supported_version) const -> std::optional<uint32_t>
+{
+    auto const lesser_version = version - 1;
```

Comment:
`lesser_version` is not a particularly meaningful variable name. Perhaps `max_allowed_version` or something would be better?